### ISSUE TITLE
pretrained CNN+GP cannot train

### ISSUE THREAD

**Issue by haohengh** [GPflow/GPflow#993]
System information:

* Python version: 3.6.8
* TensorFlow installed from (source or binary): conda-forge
* TensorFlow version (use command below): 1.12.0
* GPflow installed from (source or binary): source (pip install)
* GPflow version: 1.3.0

... _Describe the current behavior_

I'm trying to train a hybrid model with GP on top of pre-trained CNN (Densenet, VGG and Resnet) with CIFAR10 data, mimic the ex2 function in the gpflow document. But the testing result is always between 0.1~0.2, which generally means random guess. 
I've tried same code with simpler cnn models (2 conv layer or 4 conv layer) and both have reasonable results. I've tried to use different Keras applications: Densenet121, VGG16, ResNet50, neither works. I've tried to freeze the weights in the pre-trained models still not working.

... _Describe the expected behavior_
Wilson+2016 paper shows hybrid model for CIFAR10 data should get accuracy of 0.7

... _Code to reproduce the issue Provide a reproducible test case that is the bare minimum necessary to generate the problem._
```python
def cnn_dn(output_dim):
    base_model = DenseNet121(weights='imagenet', include_top=False, input_shape=(32,32,3))
    bout = base_model.output
    fcl = GlobalAveragePooling2D()(bout)
    #for layer in base_model.layers:
    #    layer.trainable = False
    output=Dense(output_dim, activation='relu')(fcl)
    md=Model(inputs=base_model.input, outputs=output)
    return md

#add gp on top, reference:ex2() function in
#https://nbviewer.jupyter.org/github/GPflow/GPflow/blob/develop/doc/source/notebooks/tailor/gp_nn.ipynb
#needs to slightly change build graph part because keras variable #sharing is not the same as tensorflow
#......

## build graph
with tf.variable_scope('cnn'):
    md=cnn_dn(gp_dim)
    f_X = tf.cast(md(X), dtype=float_type)
    f_Xtest = tf.cast(md(Xtest), dtype=float_type)

#......

    ## predict

res=np.argmax(sess.run(my, feed_dict={Xtest:xts}),1).reshape(yts.shape)
correct = res == yts.astype(int)
print(np.average(correct.astype(float)))
```
.. _Other info / logs, include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached._
posted on stackoverflow with gpflow tag, but no view or reply. 
https://stackoverflow.com/questions/56761547/pretrained-densenet-vgg16-resnet50-gp-does-not-train-on-cifar10-data

**Comment by haohengh** [comment]
Just figure out if training longer time (change number of iterations from 50 to 1000) helps solve the problem, some hyper-parameter (learning rate or activation function) adjustment also helps.
